dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -14,6 +18,13 @@ class ScriptLevelsController < ApplicationController
   # This is set to twice the proxy max-age because of a bug in CloudFront.
   DEFAULT_PUBLIC_CLIENT_MAX_AGE = DEFAULT_PUBLIC_PROXY_MAX_AGE * 2
 
+  # Return true if request is one that can be publicly cached.
+  def self.is_cachable_request?(request)
+    script_id = request.params[:script_id]
+    script = Script.get_from_cache(script_id) if script_id
+    script && Gatekeeper.allows('public_caching_for_script', where: {script_name: script.name})
+  end
```

Comment:
minor nit- consider standardizing on "cacheable" spelling for consistency with the HTTP/1.1+ RFCs?

(fun trivia around the spelling- HTTP/1.0 first used the spelling ["cachable"](https://tools.ietf.org/wg/http/draft-ietf-http-v10-spec/draft-ietf-http-v10-spec-04-from-03.diff.html) in its text, and after some [discussion](https://lists.w3.org/Archives/Public/ietf-http-wg-old/1997SepDec/0003.html) HTTP/1.1 standardized on this spelling throughout most of its development, even using a ["cachable"](https://tools.ietf.org/wg/http/draft-ietf-http-v11-spec/draft-ietf-http-v11-spec-02-from-01.diff.html) cache-control directive in early versions, until the 'e' was quietly find-replace inserted into HTTP/1.1 in a [final draft](https://tools.ietf.org/wg/http/draft-ietf-http-v11-spec-rev/draft-ietf-http-v11-spec-rev-06-from-05.diff.html).)